src/setting.rs

```diff
@@ -17,24 +17,29 @@
 //!
 //! The exact format still has to be reverse-engineered.
 
-use crate::util::nom_input_error_with_kind;
-use nom::error::ErrorKind;
-use nom::IResult;
+use binrw::{binread, NullString};
 
 #[derive(Debug)]
+#[binread]
```

Comment:
why binread instead of `binrw`?

Reply:
I wanted to archieve feature partity first. It's easy to change using search+replace, but I wasn't sure how much extra work was needed for serialization. Turns out that it's almost for free: https://github.com/Holzhaus/rekordcrate/pull/40/commits/eec1aace3a9ff18966c0ce7af847d4d1d69d5edd